### PR TITLE
ManifestBaseURL/ServiceWorkerBaseURL was missing for some paths

### PR DESCRIPTION
**Pull request by chucker**

### CODE REVIEW

**Comment by chucker** [comment]
Found another issue. The `requiredFiles` array doesn't get populated correctly either.

File: Blazor.PWA.MSBuild.Tasks/build/BlazorPWA.MSBuild.ServiceWorker.targets [hunk 1]
```diff
@@ -105,7 +105,7 @@
     <WriteLinesToFile Condition="'$(BlazorProjectType)' == 'CSB'"
                       File="$(WWWRoot)$(ServiceWorkerFileName)"
                       Overwrite="false"
-                      Lines="const requiredFiles = [;@(DistFiles -> '&quot;/%(RecursiveDir)%(FileName)%(Extension)&quot;'->Replace('\','/'),',%0D%0A');]%3B;" />
```
Comment (SQL-MisterMagoo) [inline Blazor.PWA.MSBuild.Tasks/build/BlazorPWA.MSBuild.ServiceWorker.targets:108]: This looks odd - why remove RecursiveDir? it is needed to get the full path to the file. Mistake?

**Review by SQL-MisterMagoo — COMMENTED** [review]
Can you just take a quick look at my comment on the removal of RecursiveDir?

File: Blazor.PWA.MSBuild.Tasks/build/BlazorPWA.MSBuild.ServiceWorker.targets (hunk 1 above)
Comment (chucker) [reply Blazor.PWA.MSBuild.Tasks/build/BlazorPWA.MSBuild.ServiceWorker.targets:108]: Looks like I read your proposed change in gitter wrong!

Comment (chucker) [reply Blazor.PWA.MSBuild.Tasks/build/BlazorPWA.MSBuild.ServiceWorker.targets:108]: fixed

**Comment by chucker** [comment]
Here's a sample config:

```xml
  <PropertyGroup Condition="'$(Configuration)' == 'Release'">
    <ServiceWorkerBaseUrl>/Live/</ServiceWorkerBaseUrl>
    <ManifestBaseUrl>/Live/</ManifestBaseUrl>
  </PropertyGroup>

  <PropertyGroup Condition="'$(Configuration)' == 'Staging'">
    <ServiceWorkerBaseUrl>/Staging/</ServiceWorkerBaseUrl>
    <ManifestBaseUrl>/Staging/</ManifestBaseUrl>
  </PropertyGroup>
```

I'm not sure I'm happy with the leading _and_ trailing slashes there.

**Comment by SQL-MisterMagoo** [comment]
> Here's a sample config:
> 
> ```
>   <PropertyGroup Condition="'$(Configuration)' == 'Release'">
>     <ServiceWorkerBaseUrl>/Live/</ServiceWorkerBaseUrl>
>     <ManifestBaseUrl>/Live/</ManifestBaseUrl>
>   </PropertyGroup>
> 
>   <PropertyGroup Condition="'$(Configuration)' == 'Staging'">
>     <ServiceWorkerBaseUrl>/Staging/</ServiceWorkerBaseUrl>
>     <ManifestBaseUrl>/Staging/</ManifestBaseUrl>
>   </PropertyGroup>
> ```
> 
> I'm not sure I'm happy with the leading _and_ trailing slashes there.

I'll add a couple of lines to automatically add the slashes if they are missing.

**Comment by SQL-MisterMagoo** [comment]
Do you have any projects on Github that use this package and need this feature - basically looking for something I can fork as a test project - as I don't use non-standard baseurl on anything?

If you don't - don't worry I'll make one

**Comment by chucker** [comment]
I don’t, sorry. (Can’t share the project where this came up.)

I can try and make a sample/test case tonight.

**Comment by SQL-MisterMagoo** [comment]
Thanks very much - I really appreciate your help on this.

**Pull request merged**